core/lib/instrument.c

```diff
@@ -1379,6 +1394,13 @@ instrument_fork_init(dcontext_t *dcontext)
 }
 #    endif
 
+void
+instrument_low_on_memory()
+{
+    dcontext_t *dcontext = get_thread_private_dcontext();
```

Comment:
I'm not sure about this part. How does one know whether it is safe to return drcontext to clients without fearing any race/lock problems?

Reply:
This thread owns its own dcontext.  If someone else wants to mess with it, they have to get this thread to a safe synch point using synch_with_thread.  So this is safe.  But should this callback have a dcontext parameter, when it's a global type of event where the client should free more than just the current thread's memory?

Reply:
Done.